bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -265,10 +267,50 @@ public Response getSitemapData(@Context HttpHeaders headers,
         return Response.ok(responseObject).build();
     }
 
+    /**
+     *
+     * Subscribe to a whole sitemap (all pages at once).
+     * This is not a recommended option as it incurs high SSE traffic
+     * and might pose a high load on the server, depending on the sitemap size.
+     * No built-in openhab UI should use this functionality.
+     *
+     */
     @GET
-    @Path("/{sitemapname: [a-zA-Z_0-9]+}/{pageid: [a-zA-Z_0-9]+}")
+    @Path("/{sitemapname: [a-zA-Z_0-9]+}/wholeSitemap")
```

Comment:
I would prefer something like `*`, because `wholeSitemap` is also a valid page id (at least it matchs the regexp).